manifest.js

```diff
@@ -0,0 +1,187 @@
+                let manifestSyntaxError = isValidManifest(manifestObject);
+                if(manifestSyntaxError === true){ // if manifestSyntaxError is true, i.e., there are no errors
+                    displayRulesetFilePaths(manifestObject);
+                } else {
+                    const fileInfoDiv = document.getElementById('manifestFileInfo');
+
+                    let output = "Issues found:\n";
+
+                    for (let i = 0; i < manifestSyntaxError['type'].length; i++) {
+                        if (manifestSyntaxError['type'][i] === 'missingFields') {
+                            let missingFields = manifestSyntaxError['missingFields'].join(', ');
+                            output += `- Missing fields: ${missingFields}\n`;
+                        }
+                        if (manifestSyntaxError['type'][i] === 'invalidValueTypes') {
+                            let invalidValueTypes = manifestSyntaxError['invalidValueTypes'].join(', ');
+                            output += `- Invalid value types for: ${invalidValueTypes}\n`;
+                        }
+                    }
+                    
+                    fileInfoDiv.innerText = output;
+                        
+                }
+            } catch(error){
+                console.log("Error parsing manifest JSON: ", error);
+                document.getElementById('manifestFileInfo').textContent = 'Error parsing manifest.json file';
+            }
+        }
+        reader.readAsText(manifestFile);
+    }
+});
+
+// Display the paths for the ruleset files, as defined in the manifest
+function displayRulesetFilePaths(manifest){
+    const fileInfo = document.getElementById('manifestFileInfo');
+    let output = 'Ruleset Files:\n';    
+    if(manifest.declarative_net_request.rule_resources){
+        manifest.declarative_net_request.rule_resources.forEach(ruleset => {
+            output += `- ${ruleset.id}: ${ruleset.path}, Enabled: ${ruleset.enabled}\n`;
+            rulesetFilePaths.push({rulesetFilePath: ruleset.path, rulesetId: ruleset.id});
+        });
+        // console.log(rulesetFilePaths); // correct
+        fileInfo.innerText = output;     
+    }       
+}
+
+// Checks syntax and validity of the manifest file
+function isValidManifest(manifest) {
+    let syntaxError = {};
+    syntaxError['type'] = [];
+
+    // Check for declarativeNetRequest permission
+    if(!manifest.permissions.includes('declarativeNetRequest')){
+        syntaxError.isError = true;
+        syntaxError['type'].push('missingPermissions');
+    }
+
+    // TODO: Check for declarativeNetRequestFeedback permission if debug-only features used
+    
+
+    // Check for required fields
+    const requiredFields = ['name', 'version', 'manifest_version']; // "descripton" and "icon" required for web store
+    const requiredFieldsTypes = ['string', 'string', 'number'];
+    for (let i = 0; i < requiredFields.length; i++) {
```

Comment:
If you make the above change, you should be able to just do a `for (field of requiredFields)` :)